api/nodes/filters.py

```diff
@@ -3,10 +3,36 @@
 
 from modularodm import Q
 
-from api.base.exceptions import InvalidFilterError
+from api.base.exceptions import InvalidFilterError, InvalidFilterValue
 from api.base.filters import ODMFilterMixin
 from api.base import utils
 
+from osf.models import AbstractNode as Node
+
+
+class NodesListFilterMixin(ODMFilterMixin):
+
+    def _operation_to_query(self, operation):
+        # We special case filters on root because root isn't a field; to get the children
+        # of a root, we use a custom manager method, Node.objects.get_children, and build
+        # a query from that
+        if operation['source_field_name'] == 'root':
+            child_pks = []
+            for root_guid in operation['value']:
+                if not root_guid:
+                    raise InvalidFilterValue(value=root_guid)
+                root = utils.get_object_or_error(Node, root_guid, display_name='root', prefetch_fields=self.serializer_class().model_field_names)
+                child_pks.extend(Node.objects.get_children(root=root, primary_keys=True))
+            return Q('id', 'in', child_pks)
+        elif operation['source_field_name'] == 'parent_node':
```

Comment:
`parent_nodes` is a reverse relationship on Node. If you have the guid of the parent you're looking for you can do `Node.objects.filter(parent_nodes__guids___id=parent_guid)` and that will get you a queryset of all the nodes that's parent guid is `parent_guid`. I don't know how to translate that into a MODM Q object.